lib/internal/child_process.js

```diff
@@ -516,6 +520,15 @@ ChildProcess.prototype.kill = function(sig) {
   return false;
 };
 
+ChildProcess.prototype[SymbolAsyncDispose] = function() {
+  return new Promise((resolve) => {
+    this.on('exit', () => resolve(null));
```

Comment:
What if already exited?

Reply:
What if the user remove all `exit` event handlers?